magma/magma
# chore(ci): merge bazel base and devcontainer - first pr

## Summary

This change makes the devcontainer image be build on top of the bazel base image.
* until now a lot of changes needed to be done in both images
* bazel base provides the main functionality for enabling builds with Bazel
* the devcontainer also provides QoL features for developers in vscode and also a couple of dependencies for make builds

This change is split up in 4 PRs - this is the first one
1. largest change - commits:
 a. move the bazel base image from experimental to .devcontainer - the bazel base image is not experimental anymore
 b. merge the two images and build the devcontainer image on top of bazel base. basically remove duplicated apt dependencies and change building from scratch to magma core repository dependencies when possible.
 c. build the bazel base and devcontainer image in one CI job (sequentially)
 d. do not set the latest tag for newly created bazel base and devcontainer images - set latestv2 instead. **This will cause current PRs to still use the latest images! I.e., current PRs will not fail because of not compatible images! Also jobs on master will still use latest and thus they will not fail because the new images are not created immediately.**
2. make CI workflows use latestv2 images - this will cause existing PR workflows to still use the latest tag (i.e., they will not fail) and new and rebased PRs to use the latestv2 images. This PR should be created shortly after the first PR when the image build job run successfully.
3. The image create job will use the latest tag again. This PR will be created after a couple of days when existing PRs are sufficiently created from (or rebased on) master.
4. CI workflows use the latest tag again - this can be done shortly after the third PR when the build job was run successfully.

## Test Plan

* Please note: `Build Docker image for Bazel Base and DevContainer` is expected to fail on this PR as the bazel base image is here not pushed to the magma github repo - this should succeed on master - see next bullet point for test on fork.
* run build job on fork https://github.com/nstng/magma/actions/runs/2275637718 and see that packages are created with correct tag(s) https://github.com/nstng?tab=packages&repo_name=magma (in `.devcontainer/Dockerfile` the `FROM` property was pointing temporally to fork repo)
* start vscode using the newly created devontainer - `ghcr.io/nstng/magma/devcontainer:latestv2`
* run relevant devcontainer and bazel base workflows on fork with new images (relevant for PR 2-3)
  * https://github.com/nstng/magma/actions/workflows/bazel.yml
  * https://github.com/nstng/magma/actions/workflows/gcc-problems.yml
  * https://github.com/nstng/magma/actions/workflows/bazel-cache-push.yml
  * https://github.com/nstng/magma/actions/workflows/agw-workflow.yml

## Additional Information
